src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,222 @@
+using System.Runtime.Intrinsics.X86;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.EnvironmentInfo;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Messaging;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.Datastore.Events;
+using NzbDrone.Core.Download;
+using NzbDrone.Core.Download.Clients;
+using NzbDrone.Core.Localization;
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ProviderUpdatedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ProviderDeletedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ModelEvent<RemotePathMapping>))]
+    [CheckOn(typeof(EpisodeImportFailedEvent), CheckOnCondition.SuccessfulOnly)]
+    public class RemotePathMappingCheck : HealthCheckBase, IProvideHealthCheck
+    {
+        private readonly IDiskProvider _diskProvider;
+        private readonly IProvideDownloadClient _downloadClientProvider;
+        private readonly IConfigService _configService;
+        private readonly Logger _logger;
+        private readonly IOsInfo _osInfo;
+
+        public RemotePathMappingCheck(IDiskProvider diskProvider,
+                                      IProvideDownloadClient downloadClientProvider,
+                                      IConfigService configService,
+                                      IOsInfo osInfo,
+                                      Logger logger)
+        {
+            _diskProvider = diskProvider;
+            _downloadClientProvider = downloadClientProvider;
+            _configService = configService;
+            _logger = logger;
+            _osInfo = osInfo;
+        }
+
+        public override HealthCheck Check()
+        {
+            // We don't care about client folders if we are not handling completed files
+            if (!_configService.EnableCompletedDownloadHandling)
+            {
+                return new HealthCheck(GetType());
+            }
+
+            // Only check clients not in failure status, those get another message
+            var clients = _downloadClientProvider.GetDownloadClients();
+
+            foreach (var client in clients)
+            {
+                try
+                {
+                    var status = client.GetStatus();
+                    var folders = status.OutputRootFolders;
+                    foreach (var folder in folders)
+                    {
+                        if (!folder.IsValid)
+                        {
+                            if (!status.IsLocalhost)
+                            {
+                                return new HealthCheck(GetType(), HealthCheckResult.Error, string.Format("Remote download client {0} places downloads in {1} but this is not a valid {2} path.  Review your remote path mappings and download client settings.", client.Definition.Name, folder.FullPath, _osInfo.Name), "#bad-remote-path-mapping");
```

Comment:
Same thing with double spaces here.

Reply:
Same issue here still (and others below).

Reply:
3rd times the charm; sb good now